webcompat/static/js/lib/bugform.js

```diff
@@ -341,6 +351,26 @@ function BugForm() {
     }
   };
 
+  /*
+  Check a string is a valid GitHub username
+    - maximum 39 chars
+    - alphanumerical characters and hyphens
+    - no two consecutive hyphens
+  */
+  this.isValidGitHubUsername = function(contact) {
+    return this.githubRegexp.test(contact);
+  };
+
+  /* Check to see if the GitHub username has the right syntax. */
+  this.checkGitHubUsername = function() {
+    var contact = this.contactField.val();
+    if (this.isValidGitHubUsername(contact)) {
```

Comment:
Right now, this will fail even if the input is empty, right?

Reply:
My bad, I should have given a better example. We should treat this like `osField` -- which is by default valid, but we check validation once you edit it (or something like that).

https://github.com/webcompat/webcompat.com/blob/3a261f8d6216b873001ec1e7c5a78f714fc595ab/webcompat/static/js/lib/bugform.js#L344-L352

As-is, if you never put anything in the contactField, the valid state is `null`, or if you put in a name, but then delete it all to be the empty string, it will be made invalid. I think we want the intial state to be `valid: true`.

And then when we're validating, we only make it invalid if there is a value, and it doesn't pass the regexp.

Reply:
done in https://github.com/karlcow/webcompat.com/commit/4080f289a17e8a561102a0ce403cc10bfb579ed6